docs(web-modeler): add documentation on process applications (#3786)
camunda/camunda-docs: docs/components/modeler/web-modeler/process-applications.md

This reduces the risk of having a broken deployment at runtime and makes it more convenient to deploy related files.

We advise using a process application if there is one main BPMN process that represents your end-to-end use case and
additional files the main process depends on – like called supporting processes, DMN decisions, or forms.

## Create a process application

Create a new process application by clicking the **Create process application** button in an empty project or via
**Create new** > **Process application** on the project or folder page.

<p><img src={EmptyProjectImg} alt="Create a process application from an empty project" /></p>

You will be asked to provide a name for the new process application.

<p><img src={CreateProcessApplicationImg} style={{width: 500}} alt="Create a new process application" /></p>

### Main process

The new process application will contain an empty BPMN diagram with the same name as the process application itself,
labelled as the main process.
You can rename the main process diagram any time if you wish.

:::note
A process application must always have a main process.

> That is why the main process diagram cannot be deleted or moved out of the process application.

Reviewer: This could be problem with migration. I created https://github.com/camunda/web-modeler/issues/9339 with 🔥 priority

> That is why the main process diagram cannot be deleted or moved out of the process application.

Reviewer: Here's the workaround:
"To replace a process diagram, copy-paste the BPMN XML from the desired diagram to the existing main process diagram." 
It's ugly, but it does not block the first iteration

Author: Given Valentino's comment on Slack, I've chosen this wording now:
> To replace the main process' content, copy and paste the BPMN XML from the desired diagram. Alternatively, upload a local file with the **Replace via upload** option in the breadcrumb menu or by dragging and dropping it onto the modeling canvas.

:::

### Add files to a process application

> Add more files to the process application via the **Create new** dropdown on the process application page.

Reviewer: This is a lot of text to explain basic, intuitive functionality
```suggestion
Add more files to the process application via the **Create new** dropdown on the process application page, drag and drop from your local computer, or moving from a different location in the Web Modeler.
```

You can also drag and drop files from your local computer there or move existing files from a different location in
Web Modeler to the process application.

> To make it easily discoverable, the main process will always be displayed at the top of the file list, also when you change its sort order.

Reviewer: ```suggestion
To make it easily discoverable, the main process will always be displayed at the top of the file list regardless of the sort order.
```

<p><img src={FileListImg} alt="Process application file list" /></p>

:::note
Within the same process application, process IDs, decision IDs and form IDs must be unique across all files.
This is to avoid ambiguity and conflicts when you link resources and deploy the process application later on.
:::

## Deploy and run a process application

### Deploy a process application

Use the **Deploy** button on the process application page to deploy the process application to a Zeebe cluster.

<p><img src={DeployProcessApplicationImg} alt="Deploy a process application" /></p>

> All resources contained in the process application (i.e. the main process plus all other BPMN diagrams, DMN diagrams and

Reviewer: ```suggestion
All BPMN, DMN, and form files contained in the process application folder will be deployed in a single bundle.
```

forms) will be deployed together in a single bundle.

> If any of the resources is invalid and cannot be deployed, the whole deployment will [fail](#deployment-errors) and the

Reviewer: Deployments can fail for other reasons besides a resource being invalid
```suggestion
If any resource fails to deploy, the whole deployment will [fail](#deployment-errors) and the
```

cluster state will remain unchanged.
This ensures that a process application cannot be deployed incompletely or in an inconsistent state, which makes it safer to use.

### Run a process application

> Use the **Run** button on the process application page to start a new instance of the process application's [main process](#main-process).

Reviewer: When I paired with Valentino yesterday, the **Run** button was on both the process application page and when viewing the main process in the editor page. Has that changed?

Author: @HanselIdes The **Run** button is actually displayed on the process application page and on the editor page for _every_ diagram in the process application (I just noticed that is displayed for both BPMN and DMN diagrams).

Before the actual process instance is started, all resources will be (re-)deployed if required so that the new instance
will always use their latest state.

<p><img src={RunProcessApplicationImg} alt="Run a process application" /></p>

After the process instance has been started, you will receive a notification with a link to the process instance view in
[Operate](../../operate/operate-introduction.md).

> Follow this link to observe the progress of the process instance and interact with it if required.

Reviewer: ```suggestion
Follow this link to monitor the process instance and interact with it as needed.
```

:::info

> If you click the **Deploy** or **Run** button on the modeling screen for a BPMN or DMN diagram that is part of a process

Reviewer: ```suggestion
Single file deployment is not supported in process application. If you click the **Deploy** or **Run** button on the modeling screen for any BPMN or DMN diagram in the process
```

> application, you will be deploying all resources of the process application and not just the current diagram.

Reviewer: ```suggestion
application, you will deploy all resources of the process application and not just the current diagram.
```

:::

### Deployment errors

If the deployment of a process application fails (for example, because one or more of the contained resources have invalid
implementation properties), a modal containing the error message thrown by the Zeebe engine will be displayed.
The message usually comprises the name of the affected resource, the ID of the invalid diagram element, and the actual error.

<p><img src={DeployErrorImg} style={{width: 680}} alt="Process application deployment error" /></p>

### Deployment of external resources

You can link BPMN processes, DMN decisions, or forms that are not part of the process application itself (= external
resources) from any process inside a process application.

Note that when you deploy the process application:

- linked external forms will be deployed together with the process application.
- linked external BPMN and DMN diagrams need to be deployed separately.

## Limitations and availability

> Process applications are currently a SaaS-only feature and not yet available in Web Modeler Self-Managed.

Reviewer: ```suggestion
Process applications are available in SaaS only. They are not yet available in Web Modeler Self-Managed.
```

> Also be aware of the following limitations when working with process applications:

Reviewer: This is a clear, concise list of limitations

:::note
A process application must always have a main process.
That is why the main process diagram cannot be deleted or moved out of the process application.

> To replace the main process' content, copy and paste the BPMN XML from the desired diagram.

Reviewer: This line can (and should!) be deleted. It makes me cringe to read. I'm glad there's a better option

Author: Ok, then let me delete it.

After the process instance has been started, you will receive a notification with a link to the process instance view in
[Operate](../../operate/operate-introduction.md).
Follow this link to monitor the process instance and interact with it as needed.

:::info
Single-file deployment is not supported in a process application.
If you click the **Deploy** or **Run** button on the modeling screen for any BPMN or DMN diagram in the process

> application, you will deploy _all_ resources of the process application and not just the current diagram.

Reviewer: When the redirect we described earlier today is implemented, here's a good place to mention it

Author: Yes, I will update this page again once we have implemented the change.
